website/blog/2024-12-24-the-rise-and-evolution-of-rag-in-2024-a-year-in-review/index.md

```diff
@@ -0,0 +1,296 @@
+|👎Introduces noises if the retrieved content is semantically similar but irrelevant to the answer. |                                                              |
+| 👎LLM is "indefinite" by nature.                               |                                                              |
+| 👎Difficult to consolidate the enterprise data with long context alone. |                                                              |
+| 👎Long context significantly increases both the cost of reasoning and delay. |                                                              |
+|                                                              | 👎Not smart enough: capable only of searching, unable to reason or make decisions. |
+
+Additionally, the maturation of architectures like LLMOps enabled businesses and individuals to quickly set up their own custom systems using components such as vector databases, embedding/reranking models, LLM itself, chunking tools, and prompt management techniques connected via arrows indicating data flow ensuring system usability. 
+
+![](./rag.jpg)
+
+However, applying it in broader scenarios and enterprises, and aligning its development with advancements in LLMs, still faces significant technical challenges. References [29] and [30] outline traditional academic approaches to these challenges. While some principles and practices are widely accepted, a practical analysis reveals that RAG primarily faces three main issues:
+
+1. Ineffective question answering for unstructured multimodal documents: Existing LLMOps solutions are confined to text-only scenarios. Documents such as PDFs, PowerPoint presentations (PPTs), or those integrating text with images cannot unlock their full commercial potential. These types of documents often constitute the majority within enterprise data.
+
+2. Low recall and hit rates due to pure vector databases: Relying solely on vector databases leads to low recall and hit rates, hindering effective real-world question answering. This is due to vector representations' inability to precisely represent exact information and the semantic loss during retrieval.
+
+3. The fundamental challenge of search: At its core, RAG relies on search capabilities. It works only if it can "search" for the answer based on a user's query. However, this prerequisite often fails with vague or ambiguous queries lacking clear intent or 'multi-hop' questions requiring synthesis from multiple sub-questions. In such scenarios, there is a significant semantic gap between the question posed and the answer retrieved, making traditional search methods ineffective.
+
+Therefore, the following landmark events revolve around the technical challenges of RAG:
+
+- The rise of multimodal document parsing tools.
+- The emergence of BM25 and hybrid search, rendering pure vector databases unnecessary as a separate category.
+
+### RAGFlow: Pioneering Innovations in RAG Technology
```

Comment:
RAGFlow: Pioneering Innovations in RAG Technology does not belong to the key events in RAG's evolution